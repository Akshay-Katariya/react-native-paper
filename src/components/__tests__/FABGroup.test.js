import React from 'react';
import renderer from 'react-test-renderer';
import FAB from '../FAB/FAB';

describe('FABGroup', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(
        <FAB.Group open={false} icon="add" visible={false}>
          <FAB small icon="add" />
          <FAB small icon="star" label="Star" />
          <FAB small icon="email" label="Email" />
          <FAB small icon="notifications" label="Remind" />
        </FAB.Group>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
