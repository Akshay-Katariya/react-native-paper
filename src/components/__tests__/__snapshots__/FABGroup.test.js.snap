// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FABGroup renders properly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "flex-end",
      }
    }
  >
    <View
      pointerEvents="none"
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 16,
            "marginHorizontal": 24,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "elevation": 6,
              "opacity": 0,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 6,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessibilityLabel={null}
            accessibilityRole="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 28,
                },
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, .54)",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    î……
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 16,
            "marginHorizontal": 24,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "elevation": 2,
              "marginHorizontal": 16,
              "marginVertical": 8,
              "opacity": 0,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.75,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 1.5,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <Text
              index={0}
              siblings={
                Array [
                  "withTheme(Text)",
                ]
              }
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.45999999999999996)",
                  },
                ]
              }
              total={1}
            >
              Star
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "elevation": 6,
              "opacity": 0,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 6,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessibilityLabel="Star"
            accessibilityRole="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 28,
                },
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, .54)",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    î ¸
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 16,
            "marginHorizontal": 24,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "elevation": 2,
              "marginHorizontal": 16,
              "marginVertical": 8,
              "opacity": 0,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.75,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 1.5,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <Text
              index={0}
              siblings={
                Array [
                  "withTheme(Text)",
                ]
              }
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.45999999999999996)",
                  },
                ]
              }
              total={1}
            >
              Email
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "elevation": 6,
              "opacity": 0,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 6,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessibilityLabel="Email"
            accessibilityRole="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 28,
                },
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, .54)",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    î‚¾
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 16,
            "marginHorizontal": 24,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "elevation": 2,
              "marginHorizontal": 16,
              "marginVertical": 8,
              "opacity": 0,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.75,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 1.5,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <Text
              index={0}
              siblings={
                Array [
                  "withTheme(Text)",
                ]
              }
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.45999999999999996)",
                  },
                ]
              }
              total={1}
            >
              Remind
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityTraits="button"
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "elevation": 6,
              "opacity": 0,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 6,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            accessibilityLabel="Remind"
            accessibilityRole="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 28,
                },
              ]
            }
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, .54)",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    îŸ´
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityRole="button"
      accessibilityTraits="button"
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "#03dac4",
          "borderRadius": 28,
          "elevation": 6,
          "marginBottom": 16,
          "marginHorizontal": 16,
          "marginTop": 0,
          "opacity": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 6,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 28,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, .54)",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î……
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
